src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -110,13 +111,21 @@ static HaloDBInternal open(File directory, HaloDBOptions options) throws HaloDBE
             options.getFixedKeySize(), options.getMemoryPoolChunkSize()
         );
 
-        dbInternal.buildInMemoryIndex(options);
+        long maxSequenceNumber = dbInternal.buildInMemoryIndex(options);
+
+        if (maxSequenceNumber == -1l) {
+            dbInternal.nextSequenceNumber = 0;
```

Comment:
minor change: could we start with 1 instead of 0?

Reply:
Since we are doing nextSequenceNumber++ whenever we are getting the next sequence number. It'll always start from 1 for a new db instance.